jupyter-extension/src/plugins/mount/components/Pipeline/hooks/usePipeline.tsx

```diff
@@ -130,6 +131,29 @@ export const usePipeline = (
     };
   }
 
+  const pollForIp = async (
+    project_name: string,
+    name: string,
+  ): Promise<string> => {
+    try {
+      const response = await requestAPI<PPS>(
+        `pps/${project_name}/${name}`,
+        'GET',
+      );
+      if ((response.details.service.ip !== null) && (response.details.service.ip.length > 0)) {
```

Comment:
What happens in the case that the pipeline isn't a service pipeline? Does it keep recursing infinitely?

Also, I think `response.details.service.ip` will always initially fail since the "service" key won't exist in the response on the first few calls to the endpoint. Typescript optional chaining might come in handy here.